### DISCUSSION TITLE
New release 3.9 and -core, -16, -17 dependencies

### DISCUSSION THREAD

**Discussion by pzygielo** [jqno/equalsverifier#587] [Q&A] [answered]
The newly published https://repo.maven.apache.org/maven2/nl/jqno/equalsverifier/equalsverifier/3.9/equalsverifier-3.9.pom has some new dependencies that are not available in central. All are of `provided` scope, but they cause some warnings when EV is used:
```
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (default) @ hunter ---
Downloading from central: https://repo.maven.apache.org/maven2/nl/jqno/equalsverifier/equalsverifier-core/3.9/equalsverifier-core-3.9.pom
[WARNING] The POM for nl.jqno.equalsverifier:equalsverifier-core:jar:3.9 is missing, no dependency information available
Downloading from central: https://repo.maven.apache.org/maven2/nl/jqno/equalsverifier/equalsverifier-16/3.9/equalsverifier-16-3.9.pom
[WARNING] The POM for nl.jqno.equalsverifier:equalsverifier-16:jar:3.9 is missing, no dependency information available
Downloading from central: https://repo.maven.apache.org/maven2/nl/jqno/equalsverifier/equalsverifier-17/3.9/equalsverifier-17-3.9.pom
[WARNING] The POM for nl.jqno.equalsverifier:equalsverifier-17:jar:3.9 is missing, no dependency information available
```

Can they be published or EV to undepend on them, please?

**Comment by jqno** [comment]
Is it the maven-enforcer-plugin that emits this warning? Can you show me how you've configured it? I'm trying to reproduce your issue, but I'm not seeing the error you mention.

**Reply by pzygielo** [reply]
Yes, it's enforcer plugin.

In my case - any of `dependencyConvergence`, `requireUpperBoundDeps` causes it.
Here is my plugin configuration reduced to those two rules:

```xml
            <plugin>
                <artifactId>maven-enforcer-plugin</artifactId>
                <version>3.0.0</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>enforce</goal>
                        </goals>
                        <configuration>
                            <rules>
                                <dependencyConvergence/>
                                <requireUpperBoundDeps/>
                            </rules>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

**Reply by jqno** [reply]
Thank you for the quick update! I'll see what I can do to fix this. I haven't been able to completely tame the Maven Flatten plugin yet, but I'll try some more :sweat_smile: 

In the mean time, you could consider switching to the `nl.jqno.equalsverifier:equalsverifier-nodep` artifact. It's a fat jar and contains no dependencies, and its pom is squeaky clean :smile:

**Reply by pzygielo** [reply]
Thanks for checking.
Using `equalsverifier-nodep` works perfectly! :tada:

**Reply by jqno** [reply]
Nice!

Just to check with you, it seems that adding `<optional>true</optional>` to the dependency, also seems to remove the issue. Would that solution also be sufficient for your use case? (Please stay with `-nodep` if you prefer, I just want to make the other pom file better as well :blush: )

**Reply by pzygielo** [reply]
I've only now read [CHANGELOG/2022-01-31](https://github.com/jqno/equalsverifier/blob/main/CHANGELOG.md#39---2022-01-31), and if I understand it correctly `-nodep` is kind of successor of previous `equalsverifier`, thus _works perfectly!_ is the expected outcome :smile:

> Just to check with you, it seems that adding <optional>true</optional> to the dependency, also seems to remove the issue. Would that solution also be sufficient for your use case?

:+1:

**Reply by jqno** [reply]
I've added the `<optional>` tag to the dependencies, so when the next version comes out, you can safely switch back if you prefer. Or just remain with `-nodep`, which is indeed closest to what came before.

**Reply by jqno** [reply, marked as answer]
I've just released a new version, where the dependencies are now `optional`. Just in case you want to switch back from the `-nodep` artifact :).

**Reply by pzygielo** [reply]
Thanks for the update. I confirm - new (3.9.1), no `-nodep`, EV works perfectly fine for my cases.
:tada:
